dali/pipeline/data/types.cc

```diff
@@ -20,22 +20,26 @@
 
 #define DALI_TYPEID_REGISTERER(Type, dtype)                           \
 {                                                                     \
-  std::lock_guard<std::mutex> lock(mutex_);                           \
-  static DALIDataType type_id = TypeTable::RegisterType<Type>(dtype); \
+  auto &inst = instance();                                            \
```

Comment:
```suggestion
  auto &inst = TypeTable::instance();                                            \
```
So it is clear where this `instance` comes from.